backend/pkg/connect/service.go

```diff
@@ -41,45 +41,48 @@ type ClientWithConfig struct {
 // NewService creates a new connect.Service. It tests the connectivity for each configured
 // Kafka connect cluster proactively.
 func NewService(cfg config.Connect, logger *zap.Logger) (*Service, error) {
-	logger.Info("creating Kafka connect HTTP clients and testing connectivity to all clusters")
-
 	// 1. Create a client for each configured Connect cluster
 	clientsByCluster := make(map[string]*ClientWithConfig)
-	for _, clusterCfg := range cfg.Clusters {
-		// Create dedicated Connect HTTP Client for each cluster
-		childLogger := logger.With(
-			zap.String("cluster_name", clusterCfg.Name),
-			zap.String("cluster_address", clusterCfg.URL))
-
-		opts := []con.ClientOption{
-			con.WithTimeout(cfg.ReadTimeout),
-			con.WithUserAgent("Redpanda Console"),
-		}
 
-		opts = append(opts, con.WithHost(clusterCfg.URL))
-		// TLS Config
-		tlsCfg, err := clusterCfg.TLS.TLSConfig()
-		if err != nil {
-			childLogger.Error("failed to create TLS config for Kafka connect HTTP client, fallback to default TLS config", zap.Error(err))
-			tlsCfg = &tls.Config{MinVersion: tls.VersionTLS12}
-		}
-		opts = append(opts, con.WithTLSConfig(tlsCfg))
+	if len(cfg.Clusters) > 0 {
```

Comment:
I think it would be nicer to check at the beginning af the func and return early with the dummy struct `if len(cfg.Clusters) == 0`